baseplate/lib/prometheus_metrics.py

```diff
@@ -99,6 +101,66 @@ def get_active_requests_metric(self) -> Gauge:
         return thrift_server_active_requests
 
 
+thrift_client_active_gauge = Gauge(
+    "thrift_client_active_requests", "Current in-flight requests", ["thrift_slug", "thrift_method"]
+)
+
+thrift_client_latency_histogram = Histogram(
+    "thrift_client_latency_seconds",
+    "Latency of thrift client requests",
+    ["thrift_slug", "thrift_success"],
+    buckets=default_latency_buckets,
+)
+
+thrift_client_requests_counter = Counter(
+    "thrift_client_requests_total",
+    "Total number of outgoing requests",
+    [
+        "thrift_slug",
+        "thrift_success",
+        "thrift_exception_type",
+        "thrift_baseplate_status",
+        "thrift_baseplate_status_code",
+    ],
+)
+
+
+class PrometheusThriftClientMetrics:
+    def __init__(self) -> None:
+        pass
+
+    def active_requests_metric(self, tags: Dict) -> Gauge:
+        return thrift_client_active_gauge.labels(
+            thrift_slug=tags.get("slug", ""), thrift_method=tags.get("method", "")
+        )
+
+    def requests_total_metric(self, tags: Dict) -> Counter:
+        return thrift_client_requests_counter.labels(
+            thrift_slug=tags.get("slug", ""),
```

Comment:
Sorry for being a newb here, why does this one need the labels explicitly defined when the counter has the labels already called out? Also what makes `request_total_metric` different than `get_requests_total_metric`?

Reply:
This function is useful because it translates from the tags to Prometheus labels. The required labels are declared in the metric, and this returns a counter with those labels set to values, that you then can call `.inc()` on it.

The `get_requests_total_metric` was added as a convenience to get the metric without labels and is only used in tests AFAIK. Not sure why we don't just import the metrics from the module in the test, but it's the way it was done in the server metrics.

Reply:
If this pattern already exists I'll let it slide, but I don't understand why tests need the ones without the labels (other than it maybe being more to test)

Reply:
It's because the tests don't have values to set for those labels, I think, sp they call `collect()` on the metric to get all the samples. I think this is fine. It defines a common interface to those metrics for the unit test to test all metric classes in one parameterized test.